dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v35/V2_35_11__Visualization_Move_Columns_To_Filters.java

```diff
@@ -0,0 +1,154 @@
+            final ResultSet rs = stmt.executeQuery( sql ))
+        {
+            while ( rs.next() )
+            {
+                final long visualizationId = rs.getLong( "visualizationid" );
+                final String dimension = rs.getString( "dimension" );
+                final int sortOrder = rs.getInt( "sort_order" );
+
+                // Get the greater sort_order, in filters table, for the current Visualization
+                // id.
+                int greatVisualizationSortOrder = greaterSortOrderInFiltersTableFor( context,
+                    rs.getLong( "visualizationid" ) );
+
+                // Increment the sort order so it can be inserted in the correct position.
+                greatVisualizationSortOrder++;
+
+                // Before inserting the current column into filters table, check if this column
+                // isn't already present in filters table.
+                if ( !filtersTableContains( context, visualizationId, dimension ) )
+                {
+                    // Insert the current column into filters table.
+                    insertIntoFilerTable( context, visualizationId, dimension, greatVisualizationSortOrder );
+
+                    // Once the columns is copied into filters, remove it from columns table. The
+                    // "moving" process is concluded for this visualization column.
+                    deleteFromColumnsTable( context, visualizationId, dimension, sortOrder );
+                }
+            }
+        }
+    }
+
+    private int greaterSortOrderInFiltersTableFor( final Context context, final long visualizationId )
+        throws SQLException
+    {
+        final String sql = "SELECT MAX(sort_order) AS greater_sort_order FROM visualization_filters WHERE visualizationid = ?";
+
+        try (final PreparedStatement ps = context.getConnection().prepareStatement( sql ))
+        {
+            ps.setLong( 1, visualizationId );
+
+            try (final ResultSet rs = ps.executeQuery())
+            {
+                rs.next();
+                return rs.getInt( "greater_sort_order" );
+            }
+        }
+    }
+
+    private boolean filtersTableContains( final Context context, final long visualizationId, final String dimension )
+        throws SQLException
+    {
+        final String sql = "SELECT count(visualizationid) AS vis_counter FROM visualization_filters "
+            + "WHERE visualizationid = ? AND LOWER(dimension) = LOWER(?)";
+
+        try (final PreparedStatement ps = context.getConnection().prepareStatement( sql ))
+        {
+            ps.setLong( 1, visualizationId );
+            ps.setString( 2, dimension );
+
+            try (final ResultSet rs = ps.executeQuery())
+            {
+                rs.next();
+                return rs.getInt( "vis_counter" ) > 0;
+            }
+        }
+    }
+
+    private void insertIntoFilerTable( final Context context, final long visualizationId, final String dimension,
```

Comment:
typo in method name, "Filer" not "Filter"

Reply:
Done